src/ui/public/courier/search_poll/search_poll.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+
+import { fatalError } from '../../notify';
+import '../../promises';
+import { requestQueue } from '../_request_queue';
+import { FetchSoonProvider } from '../fetch';
+
+export function SearchPollProvider(Private, Promise, $rootScope) {
+  const fetchSoon = Private(FetchSoonProvider);
+
+  class SearchPoll {
+    constructor() {
+      this._isPolling = false;
+      this._intervalInMs = undefined;
+      this._timerId = null;
+      this._searchPromise = null;
+      this._redoSearchPromise = null;
+    }
+
+    /**
+     * Set the number of milliseconds between
+     * each loop
+     *
+     * @param  {integer} intervalInMs
+     */
+    setIntervalInMs = intervalInMs => {
+      this._intervalInMs = _.parseInt(intervalInMs);
+    };
+
+    resume = () => {
+      this._isPolling = true;
+      this.resetTimer();
+    };
+
+    pause = () => {
+      this._isPolling = false;
+      this.clearTimer();
+    };
+
+    _search = () => {
+      // If we're waiting on the results of a previous search, then cancel and redo it.
```

Comment:
nit: This doesn't actually cancel the search right? It just queues another search for when this one is complete.